inventree_kicad/viewsets.py

```diff
@@ -71,16 +70,21 @@ def get_queryset(self):
 
         cascade = str2bool(plugin.get_setting('KICAD_ENABLE_SUBCATEGORY', False))
 
-        # Get the part category
-        try:
-            category = PartCategory.objects.get(id=category_id)
-        except PartCategory.DoesNotExist:
-            return Part.objects.all()
-        
-        if cascade:
-            return Part.objects.filter(category__in=category.get_descendants(include_self=True))
-        else:
-            return Part.objects.filter(category=category)
+        queryset = Part.objects.all()
```

Comment:
@SchrodingersGat this will fetch all objects at first - is that intended?

Reply:
But it doesn't evaluate yet - just construct the queryset which is augmented below